src/index.ts

```diff
@@ -1,387 +1,431 @@
 export interface ConsumerBatchOptions extends Options.Consume {
-  batching?: {
-    maxSizeBytes?: number;
-    maxTimeMs?: number;
-  };
+    batching?: {
+        maxSizeBytes?: number;
+        maxTimeMs?: number;
+    };
 }
 
 export interface IAmqpCacoonConfig {
-  protocol?: string;
-  username?: string;
-  password?: string;
-  host?: string;
-  port?: number;
-  connectionString?: string;
-  amqp_opts: object;
-  providers: {
-    logger?: Logger;
-  };
-  maxWaitForDrainMs?: number; // How long to wait for a drain event if RabbitMq fills up. Zero to wait forever. Defaults to 60000 ms (1 min)
+    protocol?: string;
+    username?: string;
+    password?: string;
+    host?: string;
+    port?: number;
+    connectionString?: string;
+    amqp_opts: object;
+    providers: {
+        logger?: Logger;
+    };
+    onChannelConnect?: ConnectCallback;
+    onBrokerConnect?: Function;
+    onBrokerDisconnect?: Function;
+    // maxWaitForDrainMs?: number; // How long to wait for a drain event if RabbitMq fills up. Zero to wait forever. Defaults to 60000 ms (1 min)
 }
+
 /**
  * AmqpCacoon
  * This module is used to communicate using the RabbitMQ amqp protocol
  * Usage
  * 1. Instantiate and pass in the configuration using the IAmqpCacoonConfig interface
  * 2. Call publish() function to publish a message
- * 3. To consume messages call registerConsumer() passing in a handler funciton
+ * 3. To consume messages call registerConsumer() passing in a handler function
  * 4. When consuming the callback registered in the previous step will be called when a message is received
  **/
 class AmqpCacoon {
-  private pubChannel: Channel | null;
-  private subChannel: Channel | null;
-  private connection?: Connection;
-  private fullHostName: string;
-  private amqp_opts: object;
-  private logger?: Logger;
-  private maxWaitForDrainMs: number;
-
-  /**
-   * constructor
-   *
-   * Usage
-   * This function just sets some class level variables
-   *
-   * @param config - Contains the amqp connection config and the Logger provder
-   **/
-  constructor(config: IAmqpCacoonConfig) {
-    this.pubChannel = null;
-    this.subChannel = null;
-    this.fullHostName = config.connectionString || this.getFullHostName(config);
-    this.amqp_opts = config.amqp_opts;
-    this.logger = config.providers.logger;
-    this.maxWaitForDrainMs = config.maxWaitForDrainMs || 60000; // Default to 1 min
-  }
-
-  /**
-   * getFullHostName
-   * Just generates the full connection name for amqp based on the passed in config
-   * @param config - Contains the amqp connection config
-   **/
-  private getFullHostName(config: IAmqpCacoonConfig) {
-    var fullHostNameString =
-      config.protocol +
-      '://' +
-      config.username +
-      ':' +
-      config.password +
-      '@' +
-      config.host;
-
-    if (config.port) {
-      fullHostNameString = fullHostNameString + ':' + config.port;
+    private pubChannelWrapper: ChannelWrapper | null;
```

Comment:
It looks like your tabwidth is set to 4 spaces instead of 2. Lets fix that for consistency :).

Reply:
hehe, uh oh...

Reply:
done